challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -2621,6 +2851,206 @@
+      ],
+      "tests": [
+        {
+          "text": "'<code>dotProduct</code> should be a function.'",
+          "testString":
+            "assert(typeof dotProduct=='function','<code>dotProduct</code> should be a function.');"
+        },
+        {
+          "text":
+            "'<code>dotProduct([1,3,-5],[4,-2,-1])</code> should return a number.'",
+          "testString":
+            "assert(typeof dotProduct([1,3,-5],[4,-2,-1]) == 'number','<code>dotProduct([1,3,-5],[4,-2,-1])</code> should return a number.');"
+        },
+        {
+          "text":
+            "'<code>dotProduct([1,3,-5],[4,-2,-1])</code> should return <code>3</code>.'",
+          "testString":
+            "assert.equal(dotProduct([1,3,-5],[4,-2,-1]),3,'<code>dotProduct([1,3,-5],[4,-2,-1])</code> should return <code>3</code>.');"
+        },
+        {
+          "text":
+            "'<code>dotProduct([1,2,3,4,5],[6,7,8,9,10])</code> should return <code>130</code>.'",
+          "testString":
+            "assert.equal(dotProduct([1,2,3,4,5],[6,7,8,9,10]),130,'<code>dotProduct([1,2,3,4,5],[6,7,8,9,10])</code> should return <code>130</code>.');"
+        },
+        {
+          "text":
+            "'<code>dotProduct([5,4,3,2],[7,8,9,6])</code> should return <code>106</code>.'",
+          "testString":
+            "assert.equal(dotProduct([5,4,3,2],[7,8,9,6]),106,'<code>dotProduct([5,4,3,2],[7,8,9,6])</code> should return <code>106</code>.');"
+        },
+        {
+          "text":
+            "'<code>dotProduct([-5,4,-3,2],[-7,-8,9,-6])</code> should return <code>-36</code>.'",
+          "testString":
+            "assert.equal(dotProduct([-5,4,-3,2],[-7,-8,9,-6]),-36,'<code>dotProduct([-5,4,-3,2],[-7,-8,9,-6])</code> should return <code>-36</code>.');"
+        },
+        {
+          "text":
+            "'<code>dotProduct([17,27,34,43,15],[62,73,48,95,110])</code> should return <code>10392</code>.'",
+          "testString":
+            "assert.equal(dotProduct([17,27,34,43,15],[62,73,48,95,110]),10392,'<code>dotProduct([17,27,34,43,15],[62,73,48,95,110])</code> should return <code>10392</code>.');"
+        }
+      ],
+      "id": "5a23c84252665b21eecc7e1e",
+      "challengeType": 5,
+      "releasedOn": "September 1, 2018",
+      "files": {
+        "indexjs": {
+          "key": "indexjs",
+          "ext": "js",
+          "name": "index",
+          "contents": [
+            "function dotProduct (ary1, ary2) {",
+            "  // Good luck!",
+            "}"
+          ],
+          "head": [],
+          "tail": []
+        }
+      }
+    },
+    {
+      "title": "Doubly-linked list/Definition",
+      "description": [
+        "Write a function that takes an array as parameter. This array contains queries that specify operations to be performed on a doubly linked list. The operations are in the form [type,args...]. The type can be head, tail or after.",
```

Comment:
"Write a function that takes an array as a parameter. This array contains queries that specify operations to be performed on a doubly linked list. The operations are in the form of <code>[ type, args... ]</code>. The type can be <code>head</code>, <code>tail</code> or <code>after</code>."